go/vt/vtgate/evalengine/compiler_asm.go

```diff
@@ -1445,6 +1445,140 @@ func (asm *assembler) Fn_RADIANS() {
 	}, "FN RADIANS FLOAT64(SP-1)")
 }
 
+func (asm *assembler) Fn_EXP() {
+	asm.emit(func(env *ExpressionEnv) int {
+		f := env.vm.stack[env.vm.sp-1].(*evalFloat)
+		f.f = math.Exp(f.f)
+		if math.IsInf(f.f, -1) || math.IsInf(f.f, 1) || math.IsNaN(f.f) {
```

Comment:
Here's an alternative way to write this, from the C stdlib's `isfinite`:

```go
func isFinite(f float64) bool {
    const mask = 0x7FF
    const shift = 64 - 11 - 1
    x := math.Float64bits(f)
    return return uint32(x>>shift)&mask != mask
}
```

Reply:
Updated!